artifactory/commands/transferfiles/longpropertycheck.go

```diff
@@ -204,15 +207,19 @@ func createSearchPropertyTask(property Property, repos []string, args precheckru
 }
 
 // Get all the files that contains the given property using AQL
-func runSearchPropertyInFilesAql(serviceManager artifactory.ArtifactoryServicesManager, property Property) (result *servicesUtils.AqlSearchResult, err error) {
+func (lpc *LongPropertyCheck) runSearchPropertyInFilesAql(serviceManager artifactory.ArtifactoryServicesManager, property Property) (result *servicesUtils.AqlSearchResult, err error) {
 	result = &servicesUtils.AqlSearchResult{}
-	err = runAqlService(serviceManager, getSearchPropertyInFilesQuery(property), result)
+	err = runAqlService(serviceManager, lpc.getSearchPropertyInFilesQuery(property), result)
 	return
 }
 
 // Get the query that search files with specific property
-func getSearchPropertyInFilesQuery(property Property) string {
-	return fmt.Sprintf(`items.find({"type": {"$eq":"any"},"@%s":"%s"}).include("repo","path","name")`, property.Key, property.Value)
+func (lpc *LongPropertyCheck) getSearchPropertyInFilesQuery(property Property) string {
+	query := fmt.Sprintf(`items.find({"type": {"$eq":"any"},"@%s":"%s"}).include("repo","path","name")`, property.Key, property.Value)
+	if lpc.disabledDistinctiveAql {
+		query += `.distinct(false)`
+	}
```

Comment:
Since the above three lines appear multiple times in the code, I sussuget putting them into an `appendDistinctIfNeeded` function, which will reduce those three lines on each of the current usages into a single line.